### PR TITLE
tests for word2vec's train(). Continuing #1139

### PR DESCRIPTION
**Pull request by robotcator**
add tests to make sure that ValueError is indeed thrown when params are not supplied.

Continuing #1139

### CODE REVIEW

File: gensim/test/test_word2vec.py [hunk 1]
```diff
@@ -637,14 +637,23 @@ def testTrainWarning(self, l):
...
+
+    def test_train_with_explicit_param(self):
+        model = word2vec.Word2Vec(size=2, min_count=1, hs=1, negative=0)
+        model.build_vocab(sentences)
+        with self.assertRaises(ValueError):
+            model.train(sentences, total_examples=model.corpus_count)
+
+        with self.assertRaises(ValueError):
+            model.train(sentences, epochs=model.iter)
+
```
Comment (tmylk) [inline gensim/test/test_word2vec.py]: please add a simple ` model.train(sentences)` example as well. Thanks

**Comment by tmylk** [comment]
Thanks. Could you please extend this PR by adding the ipynb changes?
Also please merge in develop branch to resolve the confllicts.

**Comment by robotcator** [comment]
@tmylk  Ok, I will resolve the conflicts. Since the some ipynb's codes may take several hours to complete, I  am seeking a remote server to run.

**Comment by tmylk** [comment]
Just updating the code will be fine. There is no need to re-run them.

**Comment by tmylk** [comment]
@robotcator  Are all the notebooks updated here or is it a work in progress?

**Comment by robotcator** [comment]
@tmylk all the notebooks have been updated. the commit [c9eab32](https://github.com/RaRe-Technologies/gensim/pull/1237/commits/c9eab32ab463d7edabb5e8aac5638ca1e91693bd) is the update for notebook, and i also remove my code for fixing the bug of reset_from(). I found this bug when I update the notebook, but it's duplicate in [pr1241](https://github.com/RaRe-Technologies/gensim/pull/1241).
the [8024eb5](https://github.com/RaRe-Technologies/gensim/pull/1237/commits/8024eb5182880d61d8646e98d35328a28db8ea20) is to make function testOnlineLearningAfterSave() calls use explicit count, epochs.

**Comment by tmylk** [comment]
@robotcator Thanks for finishing this PR

File: gensim/models/word2vec.py [hunk 2]
```diff
@@ -474,8 +474,8 @@ def __init__(
             if isinstance(sentences, GeneratorType):
                 raise TypeError("You can't pass a generator as the sentences argument. Try an iterator.")
             self.build_vocab(sentences, trim_rule=trim_rule)
-            self.train(sentences)
-
+            self.train(sentences, total_examples=self.corpus_count, epochs=self.iter,
```
Comment (piskvorky) [inline gensim/models/word2vec.py]: No vertical indent -- use hanging indent.

File: gensim/models/word2vec.py [hunk 3]
```diff
@@ -761,16 +761,23 @@ def _raw_word_count(self, job):
         """Return the number of words in a given job."""
         return sum(len(sentence) for sentence in job)
 
-    def train(self, sentences, total_words=None, word_count=0,
-              total_examples=None, queue_factor=2, report_delay=1.0):
+    def train(self, sentences, total_examples=None, total_words=None,
+              epochs=None, start_alpha=None, end_alpha=None,
```
Comment (piskvorky) [inline gensim/models/word2vec.py]: No vertical indent -- use hanging indent.

Comment (gojomo) [inline gensim/models/word2vec.py]: For clarity, what does 'hanging indent' mean in this context? (Does the line with `epochs=None, …` move right or left? Are lines broken differently?)

Comment (piskvorky) [inline gensim/models/word2vec.py]: Ah, this is function definition, vertical is probably OK here. I misread it as function call when reviewing.

Hanging indent would be arguments on separate lines, with extra indentation.

**Pull request merged**